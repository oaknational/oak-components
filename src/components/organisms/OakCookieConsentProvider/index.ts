export {
  OakCookieConsentProvider,
  useCookieConsent,
} from "./OakCookieConsentProvider";
