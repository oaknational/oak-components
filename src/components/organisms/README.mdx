# Organisms

## Rules for components in this folder

- Imports from atoms, molecules and occasionally organisms
- Combines multiple molecules together
- Applicable to a limited range of contexts which might occur in multiple apps
- Styled
- May contain embedded logic or state
- May be split into sub-components
  (eg. SchoolInputForm)
