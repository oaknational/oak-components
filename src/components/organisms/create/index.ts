export * from "./OakCaptionCard";
export * from "./OakCaptionSearch";
export * from "./OakCATQuestion";
