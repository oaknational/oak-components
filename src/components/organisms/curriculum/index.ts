export * from "./OakDownloadsJourneyChildSubjectTierSelector";
export * from "./OakRadioAsButton";
