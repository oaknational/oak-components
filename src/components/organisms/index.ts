export * from "./pupil";
export * from "./teacher";
export * from "./OakSubjectIcon";
export * from "./OakPrimaryNavItem";
export * from "./OakPrimaryNav";
export * from "./OakCookieBanner";
export * from "./OakCookieSettingsModal";
export * from "./OakCookieConsentProvider";
export * from "./OakCookieConsent";
