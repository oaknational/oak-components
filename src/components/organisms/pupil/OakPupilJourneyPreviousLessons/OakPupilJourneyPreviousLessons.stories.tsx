import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import { OakPupilJourneyPreviousLessons } from "./OakPupilJourneyPreviousLessons";

const meta: Meta<typeof OakPupilJourneyPreviousLessons> = {
  title: "Components/organisms/pupil/OakPupilJourneyPreviousLessons",
  component: OakPupilJourneyPreviousLessons,
  tags: ["autodocs"],
  argTypes: {},
  parameters: {
    controls: {
      include: ["numberOfLessons", "listItems"],
    },
  },
  args: {
    numberOfLessons: 5,
    listItems: ["Year 7", "Maths", "Higher"],
  },
};

export default meta;

type Story = StoryObj<typeof OakPupilJourneyPreviousLessons>;

export const Default: Story = {
  render: (args) => <OakPupilJourneyPreviousLessons {...args} />,
  args: {},
};
