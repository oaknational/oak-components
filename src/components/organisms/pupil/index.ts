export * from "./OakQuizCheckBox";
export * from "./OakQuizTextInput";
export * from "./OakQuizRadioButton";
export * from "./OakQuizHint";
export * from "./OakLessonBottomNav";
export * from "./OakLessonTopNav";
export * from "./OakQuizFeedback";
export * from "./OakQuizCounter";
export * from "./OakLessonLayout";
export * from "./OakHintButton";
export * from "./OakLessonNavItem";
export * from "./OakLessonReviewItem";
export * from "./OakLessonVideoTranscript";
export * from "./OakQuizOrder";
export * from "./OakQuizMatch";
export * from "./OakPupilJourneyLayout";
export * from "./Oakinfo";
export * from "./OakinfoButton";
export * from "./OakPupilJourneyListItem";
export * from "./OakPupilJourneyYearButton";
export * from "./OakPupilJourneySubjectButton";
export * from "./OakPupilJourneyPreviousLessons";
export * from "./OakPupilJourneyList";
export * from "./OakPupilJourneyHeader";
export * from "./OakPupilJourneyListCounter";
export * from "./OakPupilJourneyProgrammeOptions";
export * from "./OakPupilJourneyContentGuidance";
export * from "./OakPupilJourneyOptionalityItem";
export * from "./OakPupilJourneyOptionalityButton";
