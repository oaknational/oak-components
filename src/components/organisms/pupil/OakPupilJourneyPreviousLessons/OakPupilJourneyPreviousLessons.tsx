import React from "react";

import { OakBulletList, OakBulletListProps } from "@/components/molecules";
import { OakFlex } from "@/components/atoms";

export type OakPupilJourneyPreviousLessonsProps = {
  textSlot?: React.ReactNode;
} & OakBulletListProps;

/**
 * This component displays a heading for the previous lessons
 *
 *
 * listItems - List of labels to be displayed
 * textSlot? - Can pass if any react node, but <OakPupilJourneyListCounter /> is recommended
 *
 */
export const OakPupilJourneyPreviousLessons = (
  props: OakPupilJourneyPreviousLessonsProps,
) => {
  const { textSlot, listItems } = props;

  return (
    <OakFlex
      $flexDirection={["column", "row"]}
      $flexWrap={"wrap"}
      $justifyContent={"space-between"}
      $flexGrow={[null, 1]}
      $alignItems={["flex-start", "center"]}
      $gap={"space-between-m"}
    >
      {textSlot}
      <OakBulletList
        listItems={listItems}
        $background={"bg-decorative5-very-subdued"}
        $borderRadius={"border-radius-s"}
        $borderColor={"border-decorative5"}
        $ba={"border-solid-s"}
        $ph={"inner-padding-xs"}
        $pv={"inner-padding-ssx"}
      />
    </OakFlex>
  );
};
