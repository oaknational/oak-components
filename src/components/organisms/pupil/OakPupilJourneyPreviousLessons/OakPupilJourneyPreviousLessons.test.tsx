import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { OakPupilJourneyPreviousLessons } from "./OakPupilJourneyPreviousLessons";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { oakDefaultTheme } from "@/styles";
import { OakPupilJourneyListCounter } from "@/components/organisms";

describe("OakPupilJourneyPreviousLessons", () => {
  it("renders", () => {
    const { getByText } = renderWithTheme(
      <ThemeProvider theme={oakDefaultTheme}>
        <OakPupilJourneyPreviousLessons
          textSlot={
            <OakPupilJourneyListCounter
              tag="h1"
              count={5}
              countHeader="Previously released lessons"
            />
          }
          listItems={["Year 7", "Maths", "Higher"]}
          data-testid="test"
        />
      </ThemeProvider>,
    );
    expect(getByText(`Previously released lessons`)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = create(
      <ThemeProvider theme={oakDefaultTheme}>
        {" "}
        <OakPupilJourneyPreviousLessons
          textSlot={
            <OakPupilJourneyListCounter
              tag="h1"
              count={5}
              countHeader="Previously released lessons"
            />
          }
          listItems={["Year 7", "Maths", "Higher"]}
        />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
