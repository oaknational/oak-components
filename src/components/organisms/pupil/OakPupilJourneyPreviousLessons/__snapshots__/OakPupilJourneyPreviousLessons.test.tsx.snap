// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakPupilJourneyPreviousLessons matches snapshot 1`] = `
[
  " ",
  .c0 {
  margin-top: 2rem;
  font-family: Lexend,sans-serif;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: 1.5rem;
}

.c3 {
  font-family: Lexend,sans-serif;
  font-weight: 400;
  font-size: 1.25rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c4 {
  background: #fff7cc;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  border: 0.063rem solid;
  border-color: #ffe555;
  border-radius: 0.25rem;
  font-family: Lexend,sans-serif;
}

.c5 {
  font-family: Lexend,sans-serif;
}

.c2 {
  font-family: Lexend,sans-serif;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

@media (min-width:750px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:750px) {
  .c1 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:750px) {
  .c1 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media (min-width:750px) {
  .c2 {
    font-weight: 600;
  }
}

@media (min-width:750px) {
  .c2 {
    font-size: 1.25rem;
  }
}

@media (min-width:750px) {
  .c2 {
    line-height: 1.5rem;
  }
}

@media (min-width:750px) {
  .c2 {
    -webkit-letter-spacing: 0.0115rem;
    -moz-letter-spacing: 0.0115rem;
    -ms-letter-spacing: 0.0115rem;
    letter-spacing: 0.0115rem;
  }
}

<div
    className="c0 c1"
  >
    <h1
      className="c2"
    >
      Previously released lessons
      <span
        className="c3"
      >
         (5)
      </span>
    </h1>
    <span
      className="c4"
    >
      <span
        className="c5"
      >
        <span
          className="c5"
        >
          Year 7
        </span>
      </span>
      <span
        className="c5"
      >
        <span
          className="c5"
        >
           • 
        </span>
        <span
          className="c5"
        >
          Maths
        </span>
      </span>
      <span
        className="c5"
      >
        <span
          className="c5"
        >
           • 
        </span>
        <span
          className="c5"
        >
          Higher
        </span>
      </span>
    </span>
  </div>,
]
`;
